src/components/FullTextView/FullTextView.tsx

```diff
@@ -0,0 +1,87 @@
+import * as React from 'react';
+import {string, number, any, oneOfType, Requireable} from 'prop-types';
+import * as shallowequal from 'shallowequal';
+import {Tooltip} from '../Tooltip';
+import style from './FullTextView.st.css';
+
+export interface FullTextViewProps {
+  children?: React.ReactNode;
+  dataHook?: string;
+  maxWidth?: number | string;
+}
+
+export interface FullTextViewState {
+  isEllipsisActive: boolean;
+}
+
+const isEllipsisActive = (node: HTMLElement) => node && node.offsetWidth < node.scrollWidth;
+
+export class FullTextView extends React.Component<FullTextViewProps, FullTextViewState> {
+  static displayName = 'FullTextView';
+
+  static propTypes = {
+    /** dataHook to find root element */
+    dataHook: string,
```

Comment:
instead of `dataHook`, we expose it with an HOC like this:
https://github.com/wix/wix-ui-backoffice/blob/9cac9b839f2b67fe940f804131c7a34f0f9f05ec/src/components/Badge/index.ts